### PR TITLE
Update bucket creation and deletion in e2e pipeline

### PR DESCRIPTION
**Pull request by OlleLarsson**
**What this PR does / why we need it**:
Updates the pipeline to use the S3 scripts in this repository.

**Which issue this PR fixes**:

**Special notes for reviewer**:

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: .github/workflows/end-to-end.yml [hunk 1]
```diff
@@ -80,7 +58,7 @@ jobs:
       uses: actions/checkout@v2
       with:
         repository: elastisys/compliantkubernetes-apps
-        ref: main
+        ref: ol/add-s3-pipeline
```
Comment (OlleLarsson) [inline .github/workflows/end-to-end.yml:61]: Need to revert this.

File: .github/workflows/end-to-end.yml [hunk 2]
```diff
@@ -222,27 +209,14 @@ jobs:
...
       env:
         EXOSCALE_API_KEY: ${{ secrets.EXOSCALE_API_KEY }}
         EXOSCALE_API_SECRET: ${{ secrets.EXOSCALE_API_SECRET }}
       run: |
-        python3 -m venv venv
-        source venv/bin/activate
-        pip install -r S3/requirements.txt
-        ./S3/gen-s3cfg.sh exoscale "${EXOSCALE_API_KEY}" "${EXOSCALE_API_SECRET}" sos-ch-gva-2.exo.io > s3cfg.ini
-        ./S3/manager.sh --s3cfg s3cfg.ini --delete pipeline-exoscale-harbor pipeline-exoscale-velero pipeline-exoscale-es-backup pipeline-exoscale-influxdb pipeline-exoscale-sc-logs
+        ./apps/scripts/S3/generate-s3cfg.sh exoscale "${EXOSCALE_API_KEY}" "${EXOSCALE_API_SECRET}" sos-ch-gva-2.exo.io ch-gva-2 > s3cfg.ini
```
Comment (Xartos) [inline .github/workflows/end-to-end.yml:218]: Is this step needed since it's the same step as the bucket create?

**Review by Xartos — COMMENTED** [review]
Sweet! Just one minor comment

File: .github/workflows/end-to-end.yml (hunk 2 above)
Comment (OlleLarsson) [reply .github/workflows/end-to-end.yml:218]: Your right it shouldn't be needed.

File: .github/workflows/end-to-end.yml [hunk 3]
```diff
@@ -222,27 +209,13 @@ jobs:
...
-    runs-on: ubuntu-20.04
-    steps:
-    - name: Checkout bucket-manager
-      uses: actions/checkout@v2
-      with:
-        repository: elastisys/bucket-manager
-        ref: main
-        token: ${{ secrets.CK8S_PIPELINE_GITHUB_TOKEN }}
     - name: Delete buckets
+      if: matrix.cluster == 'sc'
```
Comment (lentzi90) [inline .github/workflows/end-to-end.yml:213]: I think you need to add `always()` or it won't run if there was an error earlier.
```suggestion
      if: always() && (matrix.cluster == 'sc')
```

**Review by lentzi90 — COMMENTED** [review]
Nice!

**Review by lentzi90 — APPROVED** [review]
LGTM!

**Review by Xartos — APPROVED** [review]
LGTM

**Pull request merged**